### ISSUE TITLE
error after build on upgrade TypeError: undefined is not an object (evaluating 'cppVersion.match')

### ISSUE THREAD

**Issue by ahardy42** [software-mansion/react-native-reanimated#3943]
### Description

I'm running into an error when upgrading from 1.x.x to 2.14.1.  This happens after the app builds so it's not a build error. I'm getting the following logs in the terminal that's running metro:

```
 ERROR  TypeError: undefined is not an object (evaluating 'cppVersion.match')
 ERROR  TypeError: undefined is not an object (evaluating '_$$_REQUIRE(_dependencyMap[6], "./App").default')
```

I've tried the following cleaning steps before building:

```
rm -rf node_modules
npm cache clean --force
npm install
cd ios && rm -rf Pods && pod repo update && pod install && rm -rf build && cd ..
rm -fr $TMPDIR/haste-map-*
rm -rf $TMPDIR/metro-cache
```
followed by npx react-native start --reset-cache

it looks like it's coming from the [checkVersion.ts](https://github.com/software-mansion/react-native-reanimated/tree/main/src/reanimated2/platform-specific/checkVersion.ts) file where somehow that global variable isn't being set (_REANIMATED_VERSION_CPP)

has anybody seen this before?

I'm using XCode 13.4.1

### Steps to reproduce

npm i react-native-reanimated@2.14.1
cd ios && pod install && cd ..
start metro and build for iOS

### Snack or a link to a repository

no repo (private) working on a sample repo

### Reanimated version

2.14.1

### React Native version

0.68.5

### Platforms

iOS

### JavaScript runtime

JSC

### Workflow

React Native (without Expo)

### Architecture

Paper (Old Architecture)

### Build type

Debug mode

### Device

iOS simulator

### Device model

iPhone 11 iOS 15.2

### Acknowledgements

Yes

**Comment by tomekzaw** [comment]
Hey @ahardy42, thanks for reporting this issue.

`_REANIMATED_VERSION_CPP` should be set in this line: 
https://github.com/software-mansion/react-native-reanimated/blob/270e70dd30d8f54ce303058c46b6b1109f4821c6/ios/native/REAInitializer.mm#L61
Can you please check if this line gets executed (i.e. set a breakpoint and check if it's hit)?

**Comment by tomekzaw** [comment]
Unfortunately I wasn't able to reproduce this issue on a fresh RN 0.68.5 app with JSC on iOS, it works just fine for me.

Here are my steps:
```
npx react-native@next init MyApp --version 0.68.5 --skip-install
cd MyApp
yarn add react-native-reanimated@2.14.1
cd ios && pod install && cd ..
open ios/MyApp.xcworkspace
# Add `plugins: ['react-native-reanimated/plugin']` in babel.config.js
yarn start --reset-cache
# Build the app in Xcode
```

**Comment by ahardy42** [comment]
hi @tomekzaw thanks for checking into this! I put a breakpoint at that line and it isn't getting executed before throwing that error.  I also created a repo to duplicate the issue and haven't been able to duplicate it yet. I'll keep at it.

**Comment by jordan-acosta** [comment]
I seem to be having the same issue after installing react-native-ui-lib in an existing Expo project. I tried logging from `REAInitializer.mm`, and am also not seeing the log.

I'm using:
- react-native-ui-lib@6.28.2
- react-native-reanimated@2.14.1

**Comment by crimson-med** [comment]
Same issue here on a fresh expo install.

```
npx create-expo-app --template
--> Pick Navigation TypesScript
--> Pick Name
cd Name
npm add @react-navigation/drawer
npm run android
```

In the code I simply have: 

```
import { createDrawerNavigator } from '@react-navigation/drawer';
const Drawer = createDrawerNavigator();
```
![Screenshot 2023-01-17 at 12 01 01 PM](https://user-images.githubusercontent.com/28290546/212807227-180b4e18-0273-4130-a5ae-b931363e2de5.png)

**Comment by crimson-med** [comment]
I have the same issue as this on a fresh EXPO install.
I have documented the behavior here: https://github.com/react-navigation/react-navigation/issues/11161#issuecomment-1397313189

Step by step recap:

```bash
npx create-expo-app --template
--> Pick Navigation TypesScript
--> Pick Name
cd Name
npm add @react-navigation/drawer
npm run android
```

Code: 

```ts
import { createDrawerNavigator } from '@react-navigation/drawer';
const Drawer = createDrawerNavigator();
```

You can find a reproducible repo here: https://github.com/crimson-med/react-navigation-issue

**Comment by tomekzaw** [comment]
Hey @crimson-med, do you use Expo managed or bare workflow?

**Comment by crimson-med** [comment]
> Hey @crimson-med, do you use Expo managed or bare workflow?

Expo I used the `create-expo-app`

**Comment by tomekzaw** [comment]
Yeah I know, but do you build the app yourself using `expo prebuild` or use Expo Go?

**Comment by crimson-med** [comment]
> Yeah I know, but do you build the app yourself using `expo prebuild` or use Expo Go?

I use expo go running in local on the android simulator.

**Comment by tomekzaw** [comment]
Okay, that explains everything. You cannot use other version of react-native-reanimated than the one bundled into your Expo SDK (this also applies to other native third-party libraries). That's because the JS code is newer than the native code already compiled into Expo Go. Also, in such case you should see some warning about incompatible versions when running `expo start`, is there any?

**Comment by crimson-med** [comment]
> 



> Okay, that explains everything. You cannot use other version of react-native-reanimated than the one bundled into your Expo SDK (this also applies to other native third-party libraries). That's because the JS code is newer than the native code already compiled into Expo Go. Also, in such case you should see some warning about incompatible versions when running `expo start`, is there any?

The first error i get is the one mentioned:

```
TypeError: undefined is not an object (evaluating 'cppVersion.match')
```

attached a showcase of the feed

<img width="1021" alt="Screenshot 2023-01-20 at 7 10 35 PM" src="https://user-images.githubusercontent.com/28290546/213681969-4066359d-6f9c-48c6-a992-32655e39f66a.png">


Log: 


```
 Opening exp://xxxxxx:19000 on Pixel_XL_API_30
▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
█ ▄▄▄▄▄ █   █▄▀▀▄▀█ ▄▄▄▄▄ █
█ █   █ █ ▀▄ █▀█ ▄█ █   █ █
█ ▄▄▄▄▄ █▀ ▄ ██▀▀ █▄█ ▀▀███
█ █   █ █▄ ▄▀▀▀▄█ ▄ ▄▄▀ ▀▀█
█ █▄▄▄█ █▀█▄ ███▀█▀▄▀█▀▀ ██
█▄▄▄▄▄▄▄█▄▄▄███▄██▄▄▄▄▄▄▄▄█

› Metro waiting on exp://xxxxxxxxxx:19000
› Scan the QR code above with Expo Go (Android) or the Camera app (iOS)

› Press a │ open Android
› Press i │ open iOS simulator
› Press w │ open web

› Press j │ open debugger
› Press r │ reload app
› Press m │ toggle menu

› Press ? │ show all commands

Logs for your project will appear below. Press Ctrl+C to exit.
Android Bundling complete 4584ms
 ERROR  TypeError: undefined is not an object (evaluating 'cppVersion.match')
 ERROR  Invariant Violation: Failed to call into JavaScript module method AppRegistry.runApplication(). Module has not been registered as callable. Registered callable JavaScript modules (n = 11): Systrace, JSTimers, HeapCapture, SamplingProfiler, RCTLog, RCTDeviceEventEmitter, RCTNativeAppEventEmitter, GlobalPerformanceLogger, JSDevSupportModule, HMRClient, RCTEventEmitter.
        A frequent cause of the error is that the application entry file path is incorrect. This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.
 ERROR  Invariant Violation: Failed to call into JavaScript module method AppRegistry.runApplication(). Module has not been registered as callable. Registered callable JavaScript modules (n = 11): Systrace, JSTimers, HeapCapture, SamplingProfiler, RCTLog, RCTDeviceEventEmitter, RCTNativeAppEventEmitter, GlobalPerformanceLogger, JSDevSupportModule, HMRClient, RCTEventEmitter.
        A frequent cause of the error is that the application entry file path is incorrect. This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.
```

**Comment by Latropos** [comment]
Closing since https://github.com/software-mansion/react-native-reanimated/pull/3944 is already merged
